sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpRedirectPolicy.java

```diff
@@ -91,6 +92,15 @@ public Response<BinaryData> process(HttpRequest httpRequest, HttpPipelineNextPol
         return attemptRedirect(logger, next, 0, new LinkedHashSet<>(), instrumentationContext);
     }
 
+    @Override
+    public CompletableFuture<Response<BinaryData>> processAsync(HttpRequest httpRequest, HttpPipelineNextPolicy next) {
+        // Reset the attemptedRedirectUris for each individual request.
+        InstrumentationContext instrumentationContext = httpRequest.getContext().getInstrumentationContext();
+
+        ClientLogger logger = getLogger(httpRequest);
+        return attemptRedirectAsync(logger, next, 0, new LinkedHashSet<>(), instrumentationContext);
```

Comment:
Because I'm an old man, I get a little allergic to allocations:

```suggestion
        return attemptRedirectAsync(getLogger(httpRequest), next, 0, new LinkedHashSet<>(), httpRequest.getContext().getInstrumentationContext());
```